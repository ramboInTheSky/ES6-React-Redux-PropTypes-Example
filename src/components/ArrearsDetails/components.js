import styled from 'styled-components';
import { Button, Typography, mediaQueries } from 'nhh-styles';

export const Heading = styled(Typography.H3)`
  margin-bottom: 1.5rem;
`;

export const Label = styled.span`
  font-weight: 700;
  ${mediaQueries.desktop`
    white-space: nowrap;
  `};
  ${mediaQueries.tablet`
    white-space: nowrap;
  `};
`;

export const LabelLink = styled(Button)`
  font-weight: 700;
  font-size: 18px;
  padding: 0;
  text-align: left;
`;

export const Wrapper = styled.div`
  margin-bottom: 1.5rem;
`;
