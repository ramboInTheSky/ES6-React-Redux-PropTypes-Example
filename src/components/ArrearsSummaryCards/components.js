import styled from 'styled-components';
import { mediaQueries } from 'nhh-styles';

// eslint-disable-next-line import/prefer-default-export
export const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin-top: 12px;

  > a {
    color: inherit;
    margin-bottom: 16px;
    margin-right: 1%;
    width: 32%;

    &:hover {
      color: inherit;
      text-decoration: none;
    }

    ${mediaQueries.mobile`
      margin-right: 0;
      width: 100%;
    `};

    ${mediaQueries.tablet`
      width: 49%;
    `};
  }
`;
