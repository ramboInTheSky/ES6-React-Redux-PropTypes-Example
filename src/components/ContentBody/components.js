import styled from 'styled-components';

export const OuterContainer = styled.div`
  padding-top: 50px;
  padding-bottom: 50px;
`;
export const InnerContainer = styled.div``;
