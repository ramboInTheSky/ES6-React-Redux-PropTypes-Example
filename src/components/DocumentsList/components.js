import styled from 'styled-components';
import { Button, mediaQueries, Typography } from 'nhh-styles';

export const H3 = styled(Typography.H3)`
  margin-bottom: 30px;
`;

export const Link = styled(Button)`
  font-size: inherit;
  font-weight: 400;
  display: inline;
`;

export const Container = styled.section`
  padding: 25px 28px 0;

  ${mediaQueries.mobile`
    padding: 15px 18px;
  `};
`;

export const ErrorMessage = styled.div`
  color: ${({ theme }) => (theme ? theme.colors.support.two : 'inherit')};
  margin-bottom: 50px;
`;

export const DateField = styled.div``;

export const Row = styled.div`
  border-bottom: 1px solid ${({ theme }) => theme.colors.tertiaryLighter};
  color: ${({ theme }) => theme.colors.tertiary};
  display: flex;
  flex-direction: row;
  font-size: 16px;
  padding: 8px 0;
`;

export const Header = Row.extend`
  font-size: 14px;
  font-weight: bold;
  text-transform: uppercase;
`;

export const Column = styled.div`
  width: ${({ width }) => width || 'auto'};
  text-align: ${({ align }) => align || 'left'};
  padding-right: 8px;
  word-wrap: break-word;

  &:last-child {
    padding-right: 0;
  }

  ${mediaQueries.mobile`
    padding-right: 4px;
  `};
`;

export const DesktopView = styled.div`
  display: none;

  ${mediaQueries.desktop`
    display: block;
  `};
`;

export const MobileAndTabletView = styled.div`
  display: block;

  ${mediaQueries.desktop`
    display: none;
  `};
`;
