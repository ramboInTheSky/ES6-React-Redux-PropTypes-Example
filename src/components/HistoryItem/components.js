import styled from 'styled-components';
import { Typography } from 'nhh-styles';

export const Label = styled.span`
  font-weight: 700;
`;

export const Title = styled(Typography.H3)`
  margin: -30px 0 30px 0;
`;

export const Value = styled.span`
  word-break: break-word;
`;
