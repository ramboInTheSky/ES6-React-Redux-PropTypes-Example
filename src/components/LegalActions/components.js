import styled, { withTheme } from 'styled-components';
import { Typography } from 'nhh-styles';

export const Wrapper = styled.div`
  margin: 1.5rem 0 1.875rem 0;
`;

export const P = styled(Typography.P)`
  margin: 1rem 0;
`;

export const Label = styled.span`
  font-weight: 700;
`;

export const Highlight = withTheme(styled.span`
  color: ${({ theme, isExpiring }) => (isExpiring ? theme.colors.support.two : 'inherit')};
  ${({ isExpiring }) => isExpiring && `font-weight: 700;`};
`);
