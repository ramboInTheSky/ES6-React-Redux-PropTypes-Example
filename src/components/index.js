export { default as ArrearsDetails } from './ArrearsDetails';
export { default as ArrearsSummaryCards } from './ArrearsSummaryCards';
export { default as ButtonRow } from './ButtonRow';
export { default as ContentBody } from './ContentBody';
export { default as ErrorBoundary } from './ErrorBoundary';
export { default as FieldRow } from './FieldRow';
export { default as FormWrapper } from './FormWrapper';
export { default as Heading } from './Heading';
export { default as HistoryItem } from './HistoryItem';
export { default as LegalActions } from './LegalActions';
export { default as MediaTableView } from './MediaTableView';
export { default as ModalFieldRow } from './ModalFieldRow';
export { default as NoTouching } from './NoTouching';
export { default as PreviewLinkContainer } from './PreviewLinkContainer';
export { default as PreviewTemplate } from './PreviewTemplate';
export { default as PreviewLink } from './PreviewLink';
export { default as NotificationWrapper } from './NotificationWrapper';
export { default as TaskTitle } from './TaskTitle';
export { default as TasksSummaryCards } from './TasksSummaryCards';
export { default as TemplateContainer } from './TemplateContainer';
