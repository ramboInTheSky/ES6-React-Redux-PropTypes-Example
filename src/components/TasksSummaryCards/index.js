import React from 'react';
import PropTypes from 'prop-types';
import { withTheme } from 'styled-components';
import { Link } from 'react-router-dom';
import { Card, formatting, Pagination, Loader } from 'nhh-styles';
import pathOr from 'ramda/src/pathOr';
import isAfter from 'date-fns/is_after';

import { Wrapper, OverdueLabel } from './components';
import NoDataMessage from '../../components/NoDataMessage';

const CardComponent = (task, cardLabels, theme, items, taskIsOverdue, userId, youText) => {
  const { id, name } = task.owner;
  const ownedByUser = id === userId;
  const ownerName = ownedByUser ? youText : name;
  return (
    <Card>
      <Card.Header
        color={theme.colors.secondary}
        badgeText={ownerName}
        badgeUseInitials={!ownedByUser}
      >
        {cardLabels.heading}
      </Card.Header>
      <Card.Body items={items} />
      <Card.Footer color={theme.colors.secondary}>
        {task.status}{' '}
        {taskIsOverdue ? (
          <OverdueLabel themedColor={theme.colors.support.two}> {cardLabels.overdue}</OverdueLabel>
        ) : null}
      </Card.Footer>
    </Card>
  );
};

const getCaseUrl = item => {
  if (item && item.type) {
    return `/arrears-details/${item.id}`;
  }
  return '/';
};

export const TasksSummaryCardsComponent = ({
  userId,
  youText,
  tasks,
  cardLabels,
  theme,
  loading,
  noDataMsg,
}) =>
  // eslint-disable-next-line no-nested-ternary
  loading ? (
    <Loader />
  ) : tasks.length && !loading ? (
    <Pagination
      items={tasks}
      pageSize={24}
      render={cards => (
        <Wrapper>
          {cards.map((task, idx) => {
            const taskDueDate = task.dueDate ? formatting.formatDate(task.dueDate) : null;
            const taskIsOverdue = taskDueDate ? isAfter(new Date(), task.dueDate) : false;
            const items = [
              {
                title: task.type === 'system' ? cardLabels.dueDate : cardLabels.doDate,
                value: taskDueDate || 'N/A',
              },
              {
                title: cardLabels.taskOwner,
                value: pathOr('', ['owner', 'name'], task),
              },
              {
                title: cardLabels.title,
                value: task.title,
              },
              {
                title: cardLabels.customerName,
                value: task.tenantName,
              },
              {
                title: cardLabels.address,
                value: task.assetName,
              },
            ];
            return (
              <Link
                key={task.id}
                data-bdd={`tasksSummaryCard-${idx + 1}`}
                to={getCaseUrl(task.regardingObjectModel)}
              >
                {CardComponent(task, cardLabels, theme, items, taskIsOverdue, userId, youText)}
              </Link>
            );
          })}
        </Wrapper>
      )}
    />
  ) : (
    <NoDataMessage message={noDataMsg} />
  );

export const taskCardPropTypes = {
  dueDate: PropTypes.string,
  assetName: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  tenantName: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  owner: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
  }),
  regardingObjectModel: PropTypes.shape({
    id: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
  }),
};

export const taskCardLabelPropTypes = {
  address: PropTypes.string.isRequired,
  customerName: PropTypes.string.isRequired,
  dueDate: PropTypes.string,
  taskOwner: PropTypes.string,
  title: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  overdue: PropTypes.string.isRequired,
};

TasksSummaryCardsComponent.defaultProps = {
  cardHeading: null,
  tasks: [],
};

TasksSummaryCardsComponent.propTypes = {
  cardLabels: PropTypes.shape(taskCardLabelPropTypes).isRequired,
  loading: PropTypes.bool.isRequired,
  theme: PropTypes.shape({
    colors: PropTypes.shape({
      support: PropTypes.shape({
        three: PropTypes.string,
      }),
    }).isRequired,
  }).isRequired,
  userId: PropTypes.string.isRequired,
  youText: PropTypes.string.isRequired,
  tasks: PropTypes.arrayOf(PropTypes.shape(taskCardPropTypes)),
};

CardComponent.PropTypes = {
  ...TasksSummaryCardsComponent.propTypes,
  userId: PropTypes.string.isRequired,
  youText: PropTypes.string.isRequired,
  items: PropTypes.arrayOf({
    title: PropTypes.string,
    value: PropTypes.string,
  }),
  taskIsOverdue: PropTypes.bool.isRequired,
};

export default withTheme(TasksSummaryCardsComponent);
