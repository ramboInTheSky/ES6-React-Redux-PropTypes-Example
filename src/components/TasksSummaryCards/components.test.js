import React from 'react';
import { shallow } from 'enzyme';

import 'jest-styled-components';

import { Wrapper, OverdueLabel } from './components';

describe('TasksSummaryCards components', () => {
  it('should render <Wrapper /> correctly', () => {
    expect(shallow(<Wrapper />)).toMatchSnapshot();
  });
  it('should render <OverdueLabel /> correctly', () => {
    expect(shallow(<OverdueLabel />)).toMatchSnapshot();
  });
});
