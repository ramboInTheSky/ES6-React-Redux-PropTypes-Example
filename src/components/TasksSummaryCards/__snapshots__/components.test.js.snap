// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksSummaryCards components should render <OverdueLabel /> correctly 1`] = `
.c0 {
  font-weight: bold;
  font-style: italic;
}

<span
  className="c0"
/>
`;

exports[`TasksSummaryCards components should render <Wrapper /> correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-top: 12px;
}

.c0 > a {
  color: inherit;
  margin-bottom: 16px;
  margin-right: 1%;
  width: 32%;
}

.c0 > a:hover {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (max-width:767px) {
  .c0 > a {
    margin-right: 0;
    width: 100%;
  }
}

@media (min-width:768px) and (max-width:1000px) {
  .c0 > a {
    width: 49%;
  }
}

<div
  className="c0"
/>
`;
